File: portfolio/app/src/main/java/com/example/portfolio/MainActivity.java
package com.example.portfolio;

import android.os.Bundle;
import android.widget.ListView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class MainActivity extends AppCompatActivity {

    RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.rcv_projects), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        Project[] project = {
                new Project("Project 1", "Description 1", R.drawable.calculator),
                new Project("Project 2", "Description 2", R.drawable.getting_started),
                new Project("Project 3", "Description 3", R.drawable.hungry_developer),
                new Project("Project 4", "Description 4", R.drawable.quote),
                new Project("Project 5", "Description 5", R.drawable.tape),
                new Project("Project 1", "Description 1", R.drawable.calculator),
                new Project("Project 2", "Description 2", R.drawable.getting_started),
                new Project("Project 3", "Description 3", R.drawable.hungry_developer),
                new Project("Project 4", "Description 4", R.drawable.quote),
                new Project("Project 5", "Description 5", R.drawable.tape),
                new Project("Project 1", "Description 1", R.drawable.calculator),
                new Project("Project 2", "Description 2", R.drawable.getting_started),
                new Project("Project 3", "Description 3", R.drawable.hungry_developer),
                new Project("Project 4", "Description 4", R.drawable.quote),
                new Project("Project 5", "Description 5", R.drawable.tape),
                new Project("Project 1", "Description 1", R.drawable.calculator),
                new Project("Project 2", "Description 2", R.drawable.getting_started),
                new Project("Project 3", "Description 3", R.drawable.hungry_developer),
                new Project("Project 4", "Description 4", R.drawable.quote),
                new Project("Project 5", "Description 5", R.drawable.tape),
        };

        recyclerView = (RecyclerView) findViewById(R.id.rcv_projects);
//        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        ProjectsAdapter myAdapter = new ProjectsAdapter(project);
        recyclerView.setAdapter(myAdapter);
    }
}

File: BTTH5/Bai4/app/src/main/java/com/example/bai4/MainActivity.kt
package com.example.bai4

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var editTextUrl: EditText
    private lateinit var buttonDownload: Button
    private lateinit var progressBar: ProgressBar
    private lateinit var imageView: ImageView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Liên kết với các thành phần giao diện
        editTextUrl = findViewById(R.id.editTextUrl)
        buttonDownload = findViewById(R.id.buttonDownload)
        progressBar = findViewById(R.id.progressBar)
        imageView = findViewById(R.id.imageView)

        // Xử lý sự kiện nhấn nút
        buttonDownload.setOnClickListener {
            val url = editTextUrl.text.toString().trim()
            if (url.isNotEmpty()) {
                if (url.startsWith("http://") || url.startsWith("https://")) {
                    val downloadTask = DownloadImageTask(progressBar, imageView)
                    downloadTask.execute(url)
                } else {
                    Toast.makeText(this, "URL phải bắt đầu bằng http:// hoặc https://", Toast.LENGTH_SHORT).show()
                }
            } else {
                Toast.makeText(this, "Vui lòng nhập URL", Toast.LENGTH_SHORT).show()
            }
        }
    }
}